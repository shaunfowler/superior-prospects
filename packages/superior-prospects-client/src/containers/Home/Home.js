import React, { Component } from "react";
import { Grid, Paper } from "@material-ui/core";
import Properties from "../Properties";
import Updates from "../Updates";

class Home extends Component {
  componentWillMount() {
    document.title = "Superior Prospects";
  }

  render() {
    return (
      <div className="homeView container">
        <Grid container spacing={8}>
          <Grid item xs={12} sm={12} md={3} lg={3} xl={3}>
            <Paper className="paper" elevation={1}>
              <Updates />
            </Paper>
          </Grid>
          <Grid item xs={12} sm={12} md={9} lg={9} xl={9}>
            <Paper className="paper" elevation={1}>
              <Properties />
            </Paper>
          </Grid>
        </Grid>
      </div>
    );
  }
}

export default Home;
