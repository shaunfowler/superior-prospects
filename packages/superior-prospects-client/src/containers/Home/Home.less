.homeView {
  position: relative;
  flex: 1;
  margin-top: (-64px - 54px);
}
