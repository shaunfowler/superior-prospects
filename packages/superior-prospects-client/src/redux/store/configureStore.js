import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import createHistory from "history/createBrowserHistory";
import { routerMiddleware } from "react-router-redux";
import rootReducer from "../reducers";

export const history = createHistory();

const initialState = {};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const enhancer = composeEnhancers(
  applyMiddleware(
    reduxImmutableStateInvariant(),
    thunk,
    routerMiddleware(history),
  )
);

const store = createStore(rootReducer, initialState, enhancer);

export default store;
