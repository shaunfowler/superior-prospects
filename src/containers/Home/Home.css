.homeView {
  flex: 1;
  padding-bottom: 20px;
}

.homeView .jumbotron {
  height: 200px;
  background: #fafafa;
  border-bottom: 1px solid #eee;
  margin-bottom: 20px;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;
}
