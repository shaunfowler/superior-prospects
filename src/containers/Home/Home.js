import React, { Component } from "react";
import { Link } from "react-router-dom";
import PropertyItem from "../../components/PropertyItem";
import UpdateItem from "../../components/UpdateItem";
import AddUpdateForm from "../../components/AddUpdateForm";

class Home extends Component {
  componentWillMount() {
    this.props.queryUpdates();
    this.props.queryProperties();
  }

  onDeleteUpdate = update => {
    this.props.deleteUpdate(update);
  };

  onCreateUpdate = update => {
    this.props.createUpdate(update);
  };

  render() {
    const updates = this.props.updates.list;
    const properties = this.props.properties.list;
    const { isAuthenticated } = this.props;
    return (
      <div className="homeView container ">
        <div className="columns">
          <div className="column">
            <h1 className="title">Updates</h1>
            {isAuthenticated && (
              <AddUpdateForm onCreate={this.onCreateUpdate} />
            )}
            {updates &&
              updates.map(u => (
                <UpdateItem
                  key={u._id}
                  id={u._id}
                  body={u.body}
                  created={u.created}
                  onDelete={this.onDeleteUpdate}
                  isUserAuthenticated={isAuthenticated}
                />
              ))}
          </div>
          <div className="column">
            <h1 className="title">Recently added properties</h1>
            {properties &&
              properties.map(p => (
                <PropertyItem
                  key={p._id}
                  name={p.name}
                  safeName={p.safeName}
                  description={p.description}
                />
              ))}
            <br />
            <Link to="/properties" className="button is-link">
              View all properties
            </Link>
          </div>
        </div>
      </div>
    );
  }
}

export default Home;
