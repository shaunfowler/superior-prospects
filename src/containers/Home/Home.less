.homeView {
  flex: 1;
  margin-top: 90px;

  .paper {
    padding: 20px;
  }
}
