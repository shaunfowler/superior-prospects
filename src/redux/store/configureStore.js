import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import reduxImmutableStateInvariant from "redux-immutable-state-invariant";
import createHistory from "history/createBrowserHistory";
import { routerMiddleware } from "react-router-redux";
import rootReducer from "../reducers";
import routeTrackerMiddleware from "../middleware/routeTrackerMiddleware";

export const history = createHistory();

const initialState = {};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const enhancer = composeEnhancers(
  applyMiddleware(
    reduxImmutableStateInvariant(),
    thunk,
    routerMiddleware(history),
    routeTrackerMiddleware
  )
);

const store = createStore(rootReducer, initialState, enhancer);

export default store;
