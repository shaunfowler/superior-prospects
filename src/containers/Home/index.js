import { connect } from "react-redux";
import Home from "./Home";
import {
  getUpdates,
  deleteUpdate,
  addUpdate
} from "../../redux/actions/updatesActions";

const mapStateToProps = state => {
  return { updates: state.updates };
};

const mapDispatchToProps = dispatch => {
  return {
    getUpdates: () => dispatch(getUpdates()),
    deleteUpdate: id => dispatch(deleteUpdate(id)),
    addUpdate: update => dispatch(addUpdate(update))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
