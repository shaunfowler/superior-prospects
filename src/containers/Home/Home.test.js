import React from "react";
import ReactDOM from "react-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Home from "./";

const mockStore = configureStore([thunk]);

it("renders without crashing", () => {
  const initialState = {
    updates: {
      loading: false,
      list: []
    },
    properties: {
      loading: false,
      list: []
    }
  };
  const store = mockStore(initialState);
  const div = document.createElement("div");
  ReactDOM.render(<Home store={store} />, div);
});
