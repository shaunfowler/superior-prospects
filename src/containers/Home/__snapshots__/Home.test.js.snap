// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home container renders without crashing 1`] = `
<div
  className="homeView container"
>
  <WithStyles(GridWrapper)
    container={true}
    spacing={8}
  >
    <WithStyles(GridWrapper)
      item={true}
      sm={3}
    >
      <WithStyles(Paper)
        className="paper"
        elevation={1}
      >
        <Connect(Updates) />
      </WithStyles(Paper)>
    </WithStyles(GridWrapper)>
    <WithStyles(GridWrapper)
      item={true}
      sm={9}
    >
      <WithStyles(Paper)
        className="paper"
        elevation={1}
      >
        <Connect(Properties) />
      </WithStyles(Paper)>
    </WithStyles(GridWrapper)>
  </WithStyles(GridWrapper)>
</div>
`;
